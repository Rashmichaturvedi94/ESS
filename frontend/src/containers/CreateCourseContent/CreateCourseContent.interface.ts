export interface CreateCourseContentProps {
  myProp?: boolean;
  id: number;
}
