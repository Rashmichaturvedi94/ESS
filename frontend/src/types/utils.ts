export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
