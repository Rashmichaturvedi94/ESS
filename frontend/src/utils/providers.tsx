import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
