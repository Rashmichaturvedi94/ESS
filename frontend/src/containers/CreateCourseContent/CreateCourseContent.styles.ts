import styled from 'styled-components';
import { CreateCourseContentProps } from './CreateCourseContent.interface';

export const CreateCourseContent = styled.div<Pick<CreateCourseContentProps, 'myProp'>>`
  /* Add styles here */
`;
