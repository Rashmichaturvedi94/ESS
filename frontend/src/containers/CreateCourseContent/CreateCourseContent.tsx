import React, { FC } from 'react';
import { CreateCourseContentProps } from './CreateCourseContent.interface';
export const CreateCourseContent: FC<CreateCourseContentProps> = () => {

  return (  <h3>Create course content</h3>
  );
};