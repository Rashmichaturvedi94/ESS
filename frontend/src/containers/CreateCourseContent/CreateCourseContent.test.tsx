import React from 'react';
import { render } from '../../utils/testUtils';
import { CreateCourseContent } from './CreateCourseContent';


describe('components/CreateCourseContent', () => {
  it('should render', () => {
    const { container } = render(<CreateCourseContent />);
    expect(container).toMatchSnapshot();
  });
});
