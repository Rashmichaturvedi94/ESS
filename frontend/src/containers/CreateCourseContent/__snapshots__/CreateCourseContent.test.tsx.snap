// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CreateCourseContent should render 1`] = `
<div>
  <div
    class="sc-bdnxRM fcrJXk"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <div />
      </div>
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <form
          id="createCourseContentForm"
        >
          <div
            class="MuiBox-root MuiBox-root-4"
          >
            <h2
              class="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft"
            >
              Create Course Content
            </h2>
            <h2
              class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
            >
              Title
            </h2>
            <div
              class="MuiFormControl-root MuiTextField-root sc-dlnjwi hREMJK MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="title"
                  placeholder="Title"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <h2
              class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
            >
              Description
            </h2>
            <textarea
              class="sc-hKFxyN eUmVkT"
              name="description"
              placeholder="Description"
              rows="4"
              style="height: -4px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="sc-hKFxyN eUmVkT"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0);"
              tabindex="-1"
            />
            <h2
              class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
            >
              Video Link
            </h2>
            <div
              class="MuiFormControl-root MuiTextField-root sc-dlnjwi hREMJK MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="url"
                  placeholder="url"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <h2
              class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
            >
              Thumbnail
            </h2>
            <label
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              role="button"
              style="background-color: white; color: black;"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm4.5 14H8c-1.66 0-3-1.34-3-3s1.34-3 3-3l.14.01C8.58 8.28 10.13 7 12 7c2.21 0 4 1.79 4 4h.5c1.38 0 2.5 1.12 2.5 2.5S17.88 16 16.5 16z"
                  />
                </svg>
                Attach File
                <input
                  hidden=""
                  id="img"
                  name="img"
                  type="file"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </label>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Save
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
